irc/nickserv.go

```diff
@@ -603,6 +642,27 @@ func nsSetHandler(service *ircService, server *Server, client *Client, command s
 				return
 			}
 		}
+	case "email":
+		newValue := params[1]
+		if !privileged {
+			if !nsLoginThrottleCheck(service, client, rb) {
+				return
+			}
+			var password string
+			if len(params) > 2 {
+				password = params[2]
+			}
+			errorMessage := nsConfirmPassword(server, account, password)
+			if errorMessage != "" {
+				service.Notice(rb, client.t(errorMessage))
+				return
+			}
+		}
+		munger = func(in AccountSettings) (out AccountSettings, err error) {
+			out = in
+			out.Email = newValue
```

Comment:
No verification required for the new email address?

Reply:
Were you thinking a sanity check, or an email with a confirmation code?

Reply:
Email with confirmation code. If email verification is required for new registrations, shouldn't it be required for email changes as well?

Reply:
Well, before this change, email verification was purely a safeguard (not a very strong one) against abusive registrations. It still fulfills that purpose even without confirming email changes.

It seems unnecessary to me to require a confirmation for an email change --- it's also going to complicate the UI further. (What would the commands look like?)

Reply:
Email verification implies to me that changes are verified as well. Without that, abusive users could immediately change their address to hide how they're verifying accounts, or to make it appear that someone else is responsible.

As for the UX, it can be similar to registering an account. They would get an email with the "/msg NickServ verify emailchange ..." command to complete the email change.